pkg/bpf/map_linux.go

```diff
@@ -155,35 +159,36 @@ func NewMap(name string, mapType MapType, mapKey MapKey, keySize int,
 			ValueSize:     uint32(valueSize),
 			MaxEntries:    uint32(maxEntries),
 			Flags:         flags,
-			InnerID:       innerID,
 			OwnerProgType: ProgTypeUnspec,
 		},
 		name:       path.Base(name),
 		DumpParser: dumpParser,
 	}
 }
 
-// NewPerCPUHashMap creates a new Map type of "per CPU hash" - object representing a BPF map
-// The number of cpus is used to have the size representation of a value when
-// a lookup is made on this map types.
-func NewPerCPUHashMap(name string, mapKey MapKey, keySize int, mapValue MapValue, valueSize, cpus, maxEntries int, flags uint32, innerID uint32, dumpParser DumpParser) *Map {
-	m := &Map{
+// NewMap creates a new Map instance - object representing a BPF map
+func NewMapWithInnerSpec(name string, mapType MapType, mapKey MapKey, mapValue MapValue, maxEntries int, flags uint32,
+	innerSpec *ebpf.MapSpec, dumpParser DumpParser) *Map {
+
+	keySize := reflect.TypeOf(mapKey).Elem().Size()
```

Comment:
note: Calls to `Elem()` can panic, such as if mapKey is not a pointer type.

Reply:
Ah, good observation, didn't realize. It's already been in use in NewMap for years, so we can live with this for now. The goal is to make bpf.Map generic in the short term, e.g. `Map[K MapKey, V MapValue]` so sizing can be done at compile time.